package/com.unity.formats.usd/Common/UsdEditorAnalytics.cs

```diff
@@ -0,0 +1,199 @@
+        const int k_MaxEventsPerHour = 1000;
+        const int k_MaxNumberOfElements = 100;
+        const string k_VendorKey = "unity.formats.usd";
+
+        // Universal USD Analytics
+        const string k_UsageEventName = "uUSDPackageUsage";
+
+        struct UsageAnalyticsData
+        {
+            public bool InitSucceeded;
+            public float TimeTakenS;
+        }
+
+        // USD Import Analytics
+        const string k_ImportEventName = "uUSDFileImport";
+
+        struct ImportAnalyticsData
+        {
+            public string FileExtension;
+            public float TimeTakenS;
+            public bool ImportSucceeded;
+            public bool IncludesMeshes;
+            public bool IncludesPointInstancer;
+            public bool IncludesMaterials;
+            public bool IncludesSkel;
+        }
+        public struct ImportResult
+        {
+            public bool Success;
+            public bool ContainsMeshes;
+            public bool ContainsPointInstancer;
+            public bool ContainsSkel;
+            public bool ContainsMaterials;
+
+            public static ImportResult Default => new ImportResult()
+            {
+                Success = false,
+                ContainsMeshes = false,
+                ContainsPointInstancer = false,
+                ContainsSkel = false,
+                ContainsMaterials = false
+            };
+        };
+
+        // USD Export Analytics
+        const string k_ExportEventName = "uUSDFileExport";
+
+        struct ExportAnalyticsData
+        {
+            public string FileExtension;
+            public float TimeTakenS;
+            public bool ExportSucceeded;
+            public bool OnlyOverrides;
+        }
+
+        // USD Recorder Export Analytics
+        const string k_RecorderExportEventName = "uUSDFileRecorderExport";
+
+        struct RecorderExportAnalyticsData
+        {
+            public string FileExtension;
+            public bool ExportSucceeded;
+            public bool OnlyOverrides;
+            public int FrameCount;
+        }
+
+        // key = event name
+        // value (bool) = whether the event is already registered
+        public static Dictionary<string, bool> sUsdEditorAnalyticsEvents = new Dictionary<string, bool>()
+        {
+            { k_UsageEventName, false },
+            { k_ImportEventName, false },
+            { k_ExportEventName, false },
+            { k_RecorderExportEventName, false }
+        };
+
+        static bool RegisterAnalytics(string eventName)
+        {
+# if USE_EDITOR_ANALYTICS
+            // If EditorAnalytics are disabled, don't waste cycles doing any of the set up
+            if (!EditorAnalytics.enabled)
+                return false;
+
+            if (sUsdEditorAnalyticsEvents[eventName])
+                return true;
+
+            AnalyticsResult result = EditorAnalytics.RegisterEventWithLimit(eventName, k_MaxEventsPerHour, k_MaxNumberOfElements, k_VendorKey);
```

Comment:
What are the other possible results for AnalyticsResult? Should we log if registration fails?

Reply:
The other results are the 'reasons' it can fail, eg UnsupportedPlatform. We could debug log it for the user, but I'm not sure whether they could necessarily do anything about it, and as the analytics has failed to initialize then we can't report it back to ourselves. 

Maybe it's worth Debug.Logging anyway, so that we could get slightly more informative bug reports if a user decided to report it.